crates/isograph_schema/src/add_pointers_to_supertypes.rs

```diff
@@ -56,7 +56,31 @@ impl UnvalidatedSchema {
                     Span::todo_generated(),
                 );
 
-                let condition_selection_set = vec![typename_selection];
+                let link_selection = WithSpan::new(
+                    ServerFieldSelection::ScalarField(ScalarFieldSelection {
+                        arguments: vec![],
+                        associated_data: ValidatedScalarFieldAssociatedData {
+                            location: FieldType::ClientField(
+                                match *subtype
+                                    .encountered_fields
+                                    .get(&"link".intern().into())
+                                    .expect("Expected link to exist")
```

Comment:
I don't think it's ideal that `.encountered_fields` is just a hash map. IMO it would be better to have a struct with special slots for all the special fields: `__typename`, an optional `id`, `link`, etc. and a hash map for all of the regular fields. that way, we don't have to do more assertions here.

#295